### ISSUE TITLE
aws-cpp-sdk-s3.lib in Nuget 1.5/1.6 Windows x64 static packages has wrong RuntimeLibrary linkage?

### ISSUE THREAD

**Issue by jmetzner1** [aws/aws-sdk-cpp#975]
I've got a problem with the Core and S3 libraries and static linking of x64 Windows that I can reproduce in a small console application in Visual Studio 2017 v15.8.4.

With the 1.4 libraries (exact versions below) obtained from NuGet, both the x86 and x64 versions of my program link fine with the linkage of AWSSDKCPP-Core and AWSSDKCPP-S3 set to static. If I update to 1.5 or 1.6, the exact same project links fine in x86, but in x64 has the link error:

aws-cpp-sdk-s3.lib(ub_S3.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MTd_StaticDebug' in ConsoleApplication20.obj

Is it possible that the library in the 1.5/1.6 NuGet static packages was compiled with the wrong RuntimeLibrary linkage?

**Comment by wps132230** [comment]
Hi, I am testing it with
- VS 2017 15.5.4
- AWSSDKCPP-Core 1.5.25
- AWSSDKCPP-S3 1.5.25
- Static linking
- X64

And the sample code is quite basic
```
#include <aws/core/Aws.h>
#include <aws/s3/S3Client.h>
#include <aws/core/utils/Outcome.h>

int main()
{
    Aws::SDKOptions options;
    options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Trace;
    Aws::InitAPI(options);

    Aws::S3::S3Client client;
    auto outcome = client.ListBuckets();
    if (outcome.IsSuccess())
    {
        std::cout << "SUCCESS" << std::endl;
    }
    else
    {
        std::cout << outcome.GetError().GetMessage() << std::endl;
    }

    Aws::ShutdownAPI(options);
    getchar();
    return 0;
}
```

It works for me.
Could you check both S3 and core are static linking? And what's the SDK version you are testing?

**Comment by jmetzner1** [comment]
Your example, which is simpler than my test console app, doesn't compile for me either (in a new project). Both libraries are definitely set to static linking under "Referenced Packages" in the project settings. SDK version used to compile yours and my examples is 10.0.17134.0; my original program is compiled against 10.0.16299.0.

**Comment by singku** [comment]
Hello, this problem has been fixed in 1.6.24. Please update Nuget packages and try again. Thank you!

**Comment by jmetzner1** [comment]
Works for me! Thanks!
